From 2c3708b71b2e14455a9812300519185cb1c5a70f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Alex=20R=C3=B8nne=20Petersen?= <alex@alexrp.com>
Date: Fri, 16 Jan 2026 13:25:21 +0100
Subject: [PATCH 122/133] musl: patch loongarch bits/fenv.h for soft float
X-Developer-Signature: v=1; a=openpgp-sha256; l=1609; i=xtex@astrafall.org;
 h=from:subject; bh=fRHKcfAzIeeUdFR1wRBgH0nLyCGVN7tc46L41eHWGnM=;
 b=owGbwMvMwCW2U4Ij7wZL9ETG02pJDJn78vi3iZSZb37OxCX9QqDydcNhl56LrL3SOpse3K3ce
 izsls22joksDGJcDJZiiixFhg3erDrp/KLLymVh5rAygQyRFmlgYGBgYGHgy03MKzXSMdIz1TbU
 MzTUMdIxYuDiFICpXvqCkeFC7f6/bIc4t12PDpzwVXrC/T/R3wUus6tcdb2gpdrZudON4X+d0ee
 HCybtunp9AduUB0fPHks+91o7IPHC3Xln2+7NuMXOCwA=
X-Developer-Key: i=xtex@astrafall.org; a=openpgp;
 fpr=7231804B052C670F15A6771DB918086ED8045B91

This change was missed in 1f7ee99b35b71668e32292b707213c9bf0b61c4e.

(cherry picked from commit 171b1046408bca46508ed205ff4f9a3ee4c95d80)
---
 lib/libc/include/arm-linux-musl/bits/fenv.h         | 2 +-
 lib/libc/include/loongarch64-linux-musl/bits/fenv.h | 7 ++++++-
 2 files changed, 7 insertions(+), 2 deletions(-)

diff --git a/lib/libc/include/arm-linux-musl/bits/fenv.h b/lib/libc/include/arm-linux-musl/bits/fenv.h
index 54d7b35962e8..d85fc86d7947 100644
--- a/lib/libc/include/arm-linux-musl/bits/fenv.h
+++ b/lib/libc/include/arm-linux-musl/bits/fenv.h
@@ -20,4 +20,4 @@ typedef struct {
 	unsigned long __cw;
 } fenv_t;
 
-#define FE_DFL_ENV      ((const fenv_t *) -1)
\ No newline at end of file
+#define FE_DFL_ENV      ((const fenv_t *) -1)
diff --git a/lib/libc/include/loongarch64-linux-musl/bits/fenv.h b/lib/libc/include/loongarch64-linux-musl/bits/fenv.h
index 548677f35790..31a10bb78ead 100644
--- a/lib/libc/include/loongarch64-linux-musl/bits/fenv.h
+++ b/lib/libc/include/loongarch64-linux-musl/bits/fenv.h
@@ -1,3 +1,7 @@
+#ifdef __longarch_soft_float
+#define FE_ALL_EXCEPT 0
+#define FE_TONEAREST  0
+#else
 #define FE_INEXACT    0x010000
 #define FE_UNDERFLOW  0x020000
 #define FE_OVERFLOW   0x040000
@@ -10,6 +14,7 @@
 #define FE_TOWARDZERO 0x100
 #define FE_UPWARD     0x200
 #define FE_DOWNWARD   0x300
+#endif
 
 typedef unsigned fexcept_t;
 
@@ -17,4 +22,4 @@ typedef struct {
 	unsigned __cw;
 } fenv_t;
 
-#define FE_DFL_ENV ((const fenv_t *) -1)
\ No newline at end of file
+#define FE_DFL_ENV ((const fenv_t *) -1)
-- 
2.52.0

